object/relations/relations.go

```diff
@@ -126,3 +129,230 @@ func ListRelations(ctx context.Context, rels Relations, cnrID cid.ID, rootObjID
 
 	return chain, nil
 }
+
+// HeadExecutor describes methods to get object head.
+type HeadExecutor interface {
+	ObjectHead(ctx context.Context, containerID cid.ID, objectID oid.ID, prm client.PrmObjectHead) (*client.ResObjectHead, error)
+}
+
+// SearchExecutor describes methods to search objects.
+type SearchExecutor interface {
+	ObjectSearchInit(ctx context.Context, containerID cid.ID, prm client.PrmObjectSearch) (*client.ObjectListReader, error)
+}
+
+// Executor describes all methods required to find all siblings for object.
+type Executor interface {
+	HeadExecutor
+	SearchExecutor
+}
+
+// All returns all related phy objects for provided root object ID in split-chain order.
+// If linking object is found its id will be the last one.
+//
+// Result doesn't include root object ID itself.
+func All(ctx context.Context, executor Executor, cnrID cid.ID, rootObjID oid.ID, tokens Tokens) ([]oid.ID, error) {
+	return WithLinking(ctx, executor, cnrID, rootObjID, tokens, true)
+}
+
+// WithLinking returns all related phy objects for provided root object ID in split-chain order.
+//
+// Result doesn't include root object ID itself.
+func WithLinking(ctx context.Context, executor Executor, containerID cid.ID, rootObjectID oid.ID, tokens Tokens, includeLinking bool) ([]oid.ID, error) {
```

Comment:
what is `WithLinking(..., false)`? "give me the chain with linking but... without linking"?

Reply:
It worked the same way in the previous implementation, I left it as is here. If it isn't logical, I will remove it.
Am I deleting?

Reply:
> It worked the same way in the previous implementation

oh, i see

> If it isn't logical, I will remove it.

that is strange now for sure. i would keep a single function that always returns all the relations. an additional linking object is not a big overhead

as another option, i can imagine three return values: `[]oid.ID, *oid.ID, error`. the second one is a linking object ID. that is a pointer cause it is OK to lose a linking object (well, that is a bug if we have done it but it should not be the reason for a complete object loss) and it may be missing. one more advantage for a separate linking object ID: if it is missing a user can not know about it until he looks (and analyses) at the last object in the chain according to `"If linking object is found its id will be the last one."`

Reply:
For me, three return values, in this case, are ok

Reply:
when `WithLinking` is called with `false`? i'd expect `WithLinking` / `WithoutLinking` if we really need both cases

Reply:
Old realization with false exists even in SKD [here](https://github.com/nspcc-dev/neofs-sdk-go/blob/master/audit/collect.go#L38)

Reply:
```go
func Get(ctx context.Context, executor Executor, containerID cid.ID, rootObjectID oid.ID, tokens Tokens) ([]oid.ID, *oid.ID, error)
```
covers all use-cases and pretty handsome, decided to make it so with @smallhive @carpawell